drivers/sensor/bmi270/bmi270.c

```diff
@@ -219,6 +219,7 @@ static int set_accel_odr_osr(const struct sensor_value *odr,
 		if (ret != 0) {
 			return ret;
 		}
+		k_usleep(BMI270_INTER_WRITE_DELAY_US);
```

Comment:
Is it correct to put the thread to sleep? Or will busy waiting (i.e. 	k_busy_wait ) do the same job (or even better)?

Reply:
That's a good question. Certainly going with `k_usleep()` would leave the door open for another thread to take over and have gaps between these reg-writes: `k_busy_wait()` would make the inter-write delays more deterministic.

OTOH - the inter-write delays are mostly present on single-reg configurations (e.g, set-accel/gyro-range, set-accel-odr, etc); meaning: extra-delay won't prevent them from being effective, other than a possible gap-extension between write operations.

In the end - the main reason why I chose `k_usleep()` is to keep consistency across existing delays across the same driver implementation. I don't have a strong preference: I believe both should work, but I appreciate you bringing it up.

@avisconti based on this info, do you think it's more appropriate to go with `k_busy_wait()` vs `k_usleep()`?